.github/workflows/test.yaml

```diff
@@ -24,28 +24,29 @@ jobs:
     env:
       RP_SIDEKICK_DATABASE: postgres://postgres:postgres@localhost/rp_sidekick
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
       - name: Install gettext
         run: sudo apt-get install gettext
-      - uses: actions/cache@v2
-        with:
-          path: ~/.cache/pip
-          key: ${{ hashFiles('requirements.txt', 'requirements-dev.txt') }}-pip
-      - uses: actions/setup-python@v2
+      - uses: actions/setup-python@v4
```

Comment:
I see we removed the pip cache, this [action](https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies) does have some built in cache things that supports poetry, which might speed up our builds